pkg/analyzer/analyzers/mailgun/mailgun.go

```diff
@@ -67,87 +75,60 @@ func secretInfoToAnalyzerResult(info *DomainsJSON) *analyzers.AnalyzerResult {
 	return &result
 }
 
-type Domain struct {
-	ID         string `json:"id"`
-	URL        string `json:"name"`
-	IsDisabled bool   `json:"is_disabled"`
-	Type       string `json:"type"`
-	State      string `json:"state"`
-	CreatedAt  string `json:"created_at"`
-}
-
-type DomainsJSON struct {
-	Items      []Domain `json:"items"`
-	TotalCount int      `json:"total_count"`
-}
-
-func getDomains(cfg *config.Config, apiKey string) (DomainsJSON, int, error) {
-	var domainsJSON DomainsJSON
-
-	client := analyzers.NewAnalyzeClient(cfg)
-	req, err := http.NewRequest("GET", "https://api.mailgun.net/v4/domains", nil)
+func AnalyzeAndPrintPermissions(cfg *config.Config, apiKey string) {
+	info, err := AnalyzePermissions(cfg, apiKey)
 	if err != nil {
-		return domainsJSON, -1, err
+		color.Red("[x] %s", err.Error())
+		return
 	}
 
-	req.SetBasicAuth("api", apiKey)
-	resp, err := client.Do(req)
-	if err != nil {
-		return domainsJSON, -1, err
-	}
+	color.Green("[i] Valid Mailgun API key\n\n")
+	printKeyInfo(info)
+	printDomains(info.Domains)
+	color.Yellow("[i] Permissions: Full Access\n\n")
+}
 
-	if resp.StatusCode != 200 {
-		return domainsJSON, resp.StatusCode, nil
-	}
+func AnalyzePermissions(cfg *config.Config, apiKey string) (*SecretInfo, error) {
+	var secretInfo SecretInfo
 
-	defer resp.Body.Close()
+	var client = analyzers.NewAnalyzeClient(cfg)
 
-	err = json.NewDecoder(resp.Body).Decode(&domainsJSON)
-	if err != nil {
-		return domainsJSON, resp.StatusCode, err
+	if err := getDomains(client, apiKey, &secretInfo); err != nil {
+		return &secretInfo, err
 	}
-	return domainsJSON, resp.StatusCode, nil
-}
 
-func AnalyzeAndPrintPermissions(cfg *config.Config, apiKey string) {
-	data, err := AnalyzePermissions(cfg, apiKey)
-	if err != nil {
-		color.Red("[x] %s", err.Error())
-		return
+	if err := getKeys(client, apiKey, &secretInfo); err != nil {
+		return &secretInfo, err
 	}
 
-	printMetadata(data)
+	return &secretInfo, nil
 }
 
-func AnalyzePermissions(cfg *config.Config, apiKey string) (*DomainsJSON, error) {
-	// Get the domains associated with the API key
-	domains, statusCode, err := getDomains(cfg, apiKey)
-	if err != nil {
-		return nil, fmt.Errorf("Error getting domains: %s", err)
+func printKeyInfo(info *SecretInfo) {
+	if info.ID == "" {
+		color.Red("[i] Key information not found")
+		return
 	}
 
-	if statusCode != 200 {
-		return nil, fmt.Errorf("Invalid Mailgun API key.")
-	}
-	color.Green("[i] Valid Mailgun API key\n\n")
```

Comment:
ℹ️  `AnalyzePermission()` is invoked by both `Analyze()` and `AnalyzeAndPrintPermissions()`. This function should not handle any printing; all printing should be in `AnalyzeAndPrintPermissions()`.